discovery-provider/plugins/pedalboard/apps/trending-challenge-rewards/src/trending.ts

```diff
@@ -0,0 +1,170 @@
+import App from "basekit/src/app";
+import { Knex } from "knex";
+import { SharedData } from "./config";
+import { intoResult } from "./utils";
+import { WebClient } from "@slack/web-api";
+import moment from "moment";
+import { Table, TrendingResults } from "storage/src";
+
+enum TrendingTypes {
+  Tracks = "TrendingType.TRACKS",
+  Playlists = "TrendingType.PLAYLISTS",
+  UndergroundTracks = "TrendingType.UNDERGROUND_TRACKS",
+}
+
+type TrendingEntry = {
+  handle: string; // twitter or discovery
+  rank: number;
+};
+
+export const announceTopFiveTrending = async (app: App<SharedData>, maybeWeek?: string) => {
+  const identityDbRes = await intoResult(async () => app.getIdDb());
+  if (identityDbRes.err) {
+    console.error("Identity connection error: ", identityDbRes);
+    return;
+  }
+  const identityDb = identityDbRes.unwrap();
+  const discoveryDb = app.getDnDb();
+
+  const week = maybeWeek || moment().format("YYYY-MM-DD")
+
+  const [tracks, playlists, undergroundTracks] = await queryTopFiveTrending(
+    discoveryDb,
+    week
+  );
+
+  const trackHandles = await queryHandles(identityDb, tracks);
+  const playlistHandles = await queryHandles(identityDb, playlists);
+  const undergroundHandles = await queryHandles(identityDb, undergroundTracks);
+
+  const trackEntries = assembleEntries(trackHandles, tracks);
+  const playlistEntries = assembleEntries(playlistHandles, playlists);
+  const undergroundEntries = assembleEntries(
+    undergroundHandles,
+    undergroundTracks
+  );
+
+  const trendingTracksTweet = composeTweet(
+    "Top 5 Trending Tracks 🔥",
+    week,
+    trackEntries
+  );
+  const trendingPlaylistTweet = composeTweet(
+    "Top 5 Trending Playlists 🎚️",
+    week,
+    playlistEntries
+  );
+  const trendingUndergroundTweet = composeTweet(
+    "Top 5 Trending Underground 🎵",
+    week,
+    undergroundEntries
+  );
+
+  const webClient = new WebClient(process.env.SLACK_BOT_TOKEN);
+  await sendTweet(webClient, [
+    trendingTracksTweet,
+    trendingPlaylistTweet,
+    trendingUndergroundTweet,
+  ]);
+};
+
+export const queryTopFiveTrending = async (
+  discoveryDb: Knex,
+  week: string
+): Promise<TrendingResults[][]> => {
+  // 2023-06-09
+  const tracks = await discoveryDb<TrendingResults>(Table.TrendingResults)
```

Comment:
We should try to make sure this data is the same source of truth as this data https://github.com/AudiusProject/audius-protocol/blob/bb6a26597574e65457e595d69580abd18f516266/discovery-provider/plugins/pedalboard/apps/trending-challenge-rewards/src/queries.ts#L82

But good to keep moving here without that. Just don't want us to tweet at people that end up not getting paid.